### ISSUE TITLE
upgrade to support dbt-core v1.2.0

### ISSUE THREAD

**Issue by dataders** [ClickHouse/dbt-clickhouse#79]
We've just published the release cut of dbt-core 1.2.0, `dbt-core 1.2.0rc1` ([PyPI](https://pypi.org/project/dbt-core/1.2.0rc1/) | [GitHub release notes](https://github.com/dbt-labs/dbt-core/releases/tag/v1.2.0rc1)).

https://github.com/dbt-labs/dbt-core/discussions/5468 is an open discussion with more detailed information, and https://github.com/dbt-labs/dbt-core/issues/5474 is for keeping track of the communities progress on releasing 1.2.0

Below is a checklist of work that would enable a successful 1.2.0 release of your adapter.

- [ ] migrate necessary cross-db macros into adapter and ensure they're tested accordingly
- [ ] remove any copy-and-pasted materialization (if your adapter inherits from another adapter)
- [ ] add new basic tests `BaseDocsGenerate` and `BaseDocsGenReferences`
- [ ] consider checking and testing support for Python 3.10

https://github.com/dbt-labs/dbt-core/pull/5432 might make it into the second release cut in the next week, in which case, you'll also might want to:

- [ ] implement method and tests for connection retry logic

**Comment by jesperbagge** [comment]
Would love for this to happen since table materializations fails using `dbt-core==1.2.0` and `dbt-clickhouse==1.1.7`

**Comment by genzgd** [comment]
We should get to this over the next week or two.

**Comment by genzgd** [comment]
@jesperbagge  Do you have an example of how table materializations fail when using dbt 1.2?  The basic materializations are working for me and in our automated tests, so I'd like to understand the problem.

**Comment by jesperbagge** [comment]
Well honestly, that makes us two. I created a fresh virtualenv using python3.9 and just ran `pip install dbt-clickhouse` which gave me 

```
dbt-clickhouse==1.1.7
dbt-core==1.2.1
```

And now I can't seem to be able to recreate the problem. It works like it should. I even recreated my venv with specifically targeting `dbt-core==1.2.0` but this also works as expected. I don't know what I messed up in my initial project but I'm sorry for taking up your time.
